crates/ark/src/modules/positron/methods.R

```diff
@@ -0,0 +1,56 @@
+#
+# methods.R
+#
+# Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+#
+#
+
+ark_methods_table <- new.env(parent = emptyenv())
+ark_methods_table$ark_variable_display_value <- new.env(parent = emptyenv())
+ark_methods_table$ark_variable_display_type <- new.env(parent = emptyenv())
+ark_methods_table$ark_variable_has_children <- new.env(parent = emptyenv())
+ark_methods_table$ark_variable_kind <- new.env(parent = emptyenv())
+lockEnvironment(ark_methods_table, TRUE)
+
+#' Register the methods with the Positron runtime
+#'
+#' @param generic Generic function name as a character to register
+#' @param class Class name as a character
+#' @param method A method to be registered. Should be a call object.
+#' @export
+.ps.register_ark_method <- function(generic, class, method) {
+    stopifnot(
+        typeof(generic) == "character",
+        length(generic) == 1,
+        generic %in% c(
+            "ark_variable_display_value",
+            "ark_variable_display_type",
+            "ark_variable_has_children",
+            "ark_variable_kind"
+        ),
+        typeof(class) == "character"
+    )
+    for (cls in class) {
+        assign(cls, method, envir = ark_methods_table[[generic]])
+    }
+    invisible()
+}
+
+call_ark_method <- function(generic, object, ...) {
+    methods_table <- ark_methods_table[[generic]]
+
+    if (is.null(methods_table)) {
+        return(NULL)
+    }
+
+    for (cl in class(object)) {
+        if (!is.null(method <- get0(cl, envir = methods_table))) {
+            return(eval(
+                as.call(list(method, object, ...)),
+                envir = globalenv()
+            ))
```

Comment:
As it's currently implemented `method` is not necessarily a function. It could be a call object, a symbol or a function.

Reply:
A symbol? To be evaluated in global? hmmm

I think we should evaluate the input in a child of base then, and bind to a mask to avoid the inlining.

Are you sure that symbolic objects are not evaluated on the Rust side at the `call_in()` site? Are they protected from evaluation?

Reply:
In theory, a user could do 

```
.ark.register_method("positron_display_values", "foo", quote(my_method_name)))
```

and that would call `my_method_name(x)` in the global env when dispatched.
I'm not sure I fully understand the risks of this approach.